bitcoin/src/blockdata/locktime/relative.rs

```diff
@@ -71,10 +71,17 @@ impl LockTime {
     /// Returns true if satisfaction of `other` lock time implies satisfaction of this
     /// [`relative::LockTime`].
     ///
+    /// A lock time can only be satisfied by n blocks being mined or n seconds passing. If you have
+    /// two lock times (same unit) then the larger lock time being satisfied implies (in a
+    /// mathematical sense) the smaller one being satisfied.
+    ///
     /// This function is useful when checking sequence values against a lock, first one checks the
     /// sequence represents a relative lock time by converting to `LockTime` then use this function
     /// to see if satisfaction of the newly created lock time would imply satisfaction of `self`.
     ///
+    /// Can also be used to remove the smaller value of two `OP_CHECKSEQUENCEVERIFY` operations
```

Comment:
Really a niche application. Perhaps unnecessary to write this. I can't think of any setting where someone would use two timelocks in the same branch.

Reply:
Isn't miniscript supposed to do optimizations like this?

Reply:
I agree with Sanket. This seems really niche and took me a long time to think about what situations it would be done. (As near as I can tell, only if you had a conjunction of two locktimes for some reason, but didn't want this.)

@Kixunil miniscript does not do optimizations like this, though I suppose that it could ... but even if you were eliminating locktimes it's not obvious that you'd always remove the smaller one.